### DISCUSSION TITLE
tailwindcss @apply in styled-components

### DISCUSSION THREAD

**Discussion by DanielBailey-web** [tailwindlabs/tailwindcss#7959] [Help]
Is it possible to do something like this using styled-components in tandem with tailwindcss?

```js
const Content = styled(SelectPrimitive.Content)`
  @layer components {
    [aria-disabled="true"] {
      @apply bg-black opacity-50;
    }
  }
`;
```

**Comment by CFarhad** [comment]
hi
This code not working, because the styled component can not understand `@apply` rule .
You can install [tailwind-styled-components](https://www.npmjs.com/package/tailwind-styled-components)

I'm not sure you can use the `@apply` rule in that but normally it working

**Reply by DanielBailey-web** [reply]
Yeah, it works in a normal css file, was hoping I wouldn't need to install another package haha
